File: src/server/Hoard.Core/Application/Portfolios/PortfolioExposureDto.cs
namespace Hoard.Core.Application.Portfolios;

public sealed class PortfolioExposureDto
{
    public int PortfolioId { get; init; }
    public DateOnly AsOfDate { get; init; }

    public decimal TotalValue { get; init; }

    public IReadOnlyList<AssetClassExposureDto> AssetClasses { get; init; } = [];

    public IReadOnlyList<AssetSubclassExposureDto> AssetSubclasses { get; init; } = [];
    
    public IReadOnlyList<RebalanceActionDto> RebalanceActions { get; init; } = [];
}

File: src/client/hoard-ui/src/api/dtos/PortfolioExposureDto.ts
import type { AssetClassExposureDto } from "./AssetClassExposureDto";
import type { AssetSubclassExposureDto } from "./AssetSubclassExposureDto";
import type { RebalanceActionDto } from "./RebalanceActionDto";

export interface PortfolioExposureDto {
  assetClasses: AssetClassExposureDto[];
  assetSubclasses: AssetSubclassExposureDto[];
  rebalanceActions: RebalanceActionDto[];
}
